modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/bulk/selection/action/EditCategoriesBulkSelectionAction.java

```diff
@@ -97,14 +98,21 @@ public void execute(
 			});
 	}
 
-	private long[] _getLongArray(Map<String, Serializable> map, String key) {
-		return ArrayUtil.toArray((Long[])map.getOrDefault(key, new Long[0]));
-	}
-
 	private Set<Long> _toLongSet(Map<String, Serializable> map, String key) {
 		try {
-			return SetUtil.fromArray(
-				(Long[])map.getOrDefault(key, new Long[0]));
+			Serializable values = map.get(key);
+
+			if (values instanceof Integer[]) {
+				Set<Long> set = new HashSet<>();
+
+				for (Integer value : (Integer[])values) {
+					set.add(value.longValue());
+				}
```

Comment:
Can you use `Collections.addAll` here?

Reply:
I don't believe so since they are two different types and we need to convert the Integer to Long. Let me know if I'm misunderstanding though.

Reply:
You're right :+1: